src/formatDate.js

```diff
@@ -50,7 +50,37 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const finalDateArr = [];
+  const newDate = date.split(fromFormat[3]);
+
+  if (toFormat[toFormat.lastIndexOf('YYYY')] !== undefined) {
+    let tempDate = newDate[fromFormat.lastIndexOf('YY')];
+
+    if (tempDate >= 30) {
+      tempDate = Number(tempDate) + 1900;
+    } else {
+      tempDate = Number(tempDate) + 2000;
+    }
+
+    newDate[fromFormat.lastIndexOf('YY')] = tempDate;
+    fromFormat[fromFormat.lastIndexOf('YY')] = 'YYYY';
+  } else if (toFormat[toFormat.lastIndexOf('YY')] !== undefined
+  && fromFormat[fromFormat.lastIndexOf('YYYY')] !== undefined) {
+    const tempDate = newDate[fromFormat.lastIndexOf('YYYY')].slice(2, 4);
+
```

Comment:
Create variables for long values or\and values that are repeated more than once to make your code cleaner